leshan-core/src/main/java/org/eclipse/leshan/SecurityMode.java

```diff
@@ -12,14 +12,15 @@
  * 
  * Contributors:
  *     Sierra Wireless - initial API and implementation
+ *     Rikard Höglund (RISE SICS) - Additions to support OSCORE
  *******************************************************************************/
 package org.eclipse.leshan;
 
 /**
  * The different DTLS security modes
  */
 public enum SecurityMode {
-    PSK(0), RPK(1), X509(2), NO_SEC(3);
+    PSK(0), RPK(1), X509(2), NO_SEC(3), OSCORE(4);
```

Comment:
Regarding the [specification](http://www.openmobilealliance.org/release/LightweightM2M/V1_1_1-20190617-A/HTML-Version/OMA-TS-LightweightM2M_Core-V1_1_1-20190617-A.html#Table-E1-1-LwM2M-Object-LWM2M-Security-object-definition), 4 is for _"Certificate mode with EST"_ and there is nothing for OSCORE :thinking: .

(It is not excluded there is issue in the specification but for now, I don't understand all of this enough to say if this is an issue :grin:)

Reply:
I now removed the SecurityMode for OSCORE. Instead the OSCORE Security Mode link in the Security object will be used. Done in commit 3a4e777eb664aa126c9c4bae1eb1effba7eace0e